### DISCUSSION TITLE
First Profile lost + Failure in Extension Initialization

### DISCUSSION THREAD

**Discussion by zFernand0** [zowe/zowe-explorer-vscode#1106] [Q&A]
Greetings to all the seniors,

An Absolute Beginner is here seeking guidance: VScode-Zowe on ubuntu
Learning/Playing with MTM2020 challenges.

I created a profile, and at some point when I was looking for a file and couldn't find besides I saw a message "you have not opened a folder yet", didn't know what is this for.

Created a folder [it was on local HD] with the same name mtm2020, and after that lost my profile. 

So, I deleted that folder and right after that, I can not have Zowe's initial layout.

**Reinstalled VScode several times, but Zowe comes out with no-look and the PLUS sign is not functioning [to me it seem like this], I am unable to access my profile and everything.**

PLEASE guide me on how to solve this and connect with the MTM.

Any help will be highly appreciated. Thanks

_Originally posted by @3TInfoTinker in https://github.com/zowe/vscode-extension-for-zowe/issues/829#issuecomment-743134625_

**Comment by 3TInfoTinker** [comment]
@zFernand0 , Thank you for responding to my issue

**Reply by zFernand0** [reply]
Hey @3TInfoTinker, 

- Seems like you were able to use the profile to and started looking for files. 
- And, somewhere along the way, you got an error message talking about a folder that was not opened.
- Then, you tried creating a folder/file manually in your local PC with the same name (as your profile?)
- After doing so, you couldn't see any profiles loaded on the Zowe Explorer

Please feel free to correct me if I got it completely wrong 😋 

Questions:
- Did you try reloading the window via the command palette? (`Ctrl/Cmd + Shfit + P` and type `Reload Window`)
- Where did you create the folder/file? (was it under a `~/.zowe/` folder in your `homepath`?)

**Reply by 3TInfoTinker** [reply]
I don recognize this Ctrl/CMD I'm using a HP/Desktop PC 
and the folder was created in the home directory root

**Reply by zFernand0** [reply]
Are you using Windows or *nix/Linux system?
If you close and reopen VSCode, do you get any error messages right at the start?

**Reply by 3TInfoTinker** [reply]
the folder was created in the home directory in open and also I can not find Zowe, using $pip show zowe
the system doesn't have it, 
and when I tried to install it, received an error, which also couldn't be solved.

**Reply by 3TInfoTinker** [reply]
Linux/ubuntu

**Reply by 3TInfoTinker** [reply]
no errors in running the vscode

**Reply by zFernand0** [reply]
The Zowe command line can be installed via `npm`. (see https://www.npmjs.com/package/@zowe/cli)
It should not be required for you to use the Zowe Explorer though.

Could you confirm that your profile is still under `~/.zowe/profiles/`?

**Reply by 3TInfoTinker** [reply]
I took the given command and run, 
yes It display under the main folder/zosmf/
my profile is there,
Wow, for the first time I'm seeing it

**Reply by 3TInfoTinker** [reply]
but it looks like this  :   !mtm2020.yaml

**Reply by 3TInfoTinker** [reply]
"It should not be required for you to use the Zowe Explorer though."

Actually it is required by the IBMz's MTM learning system. and I'm playing with the MTM-challenges.

**Reply by zFernand0** [reply]
So, on VSCode, when you click the Zowe Explorer icon (on the left panel) 
![image](https://user-images.githubusercontent.com/37381190/101909323-5192cc00-3b8b-11eb-8b51-a8551614a4b1.png)
Do you see the three views (`Data Sets`, `USS`, `Jobs`) and their empty `Favorites` folder?
But you do not see your `mtm2020` profile being loaded.

Could you try clicking the `+` icon on the view you are interested in (USS for instance)?
Does the mtm2020 profile show up?

**Reply by 3TInfoTinker** [reply]
The Zowe still not showing this profile, and any other btns, and the Plus btn is not responding

**Reply by 3TInfoTinker** [reply]
shall i close and run vscode again?

**Reply by zFernand0** [reply]
Please do, and let me know if you see any errors when it starts back up

**Reply by 3TInfoTinker** [reply]
ok, in a minute

**Comment by 3TInfoTinker** [comment]
No it's still the same.
![image](https://user-images.githubusercontent.com/74275625/101909997-2c7c7880-3be0-11eb-81ea-07c987c1092f.png)

**Reply by zFernand0** [reply]
An no notifications on the bottom right corner?

**Reply by 3TInfoTinker** [reply]
and profile doesn't show by clicking on the PLUS

**Reply by 3TInfoTinker** [reply]
No, just bell, and it has nothing

**Reply by zFernand0** [reply]
your profile name is `mtm2020.yaml` or `!mtm2020.yaml`?
just curious 😋

**Reply by 3TInfoTinker** [reply]
one thing, the setting's gear is showing [1] in blue

**Reply by 3TInfoTinker** [reply]
it was mtm2020

**Reply by 3TInfoTinker** [reply]
[1] Manage - New code update available.

on the leftmost bottom gear

**Reply by zFernand0** [reply]
Ok, I'll try to go through the MTM challenges myself in a minute to see if I can reproduce it on a ubuntu VM 😋

**Reply by 3TInfoTinker** [reply]
FYI, for the first time entered vscode, and likewise everything in it.

0 level beginner in all aspects, learning, leniux, ubuntu, python and now vscode 🙈

**Reply by 3TInfoTinker** [reply]
> Ok, I'll try to go through the MTM challenges myself in a minute to see if I can reproduce it on a ubuntu VM

I couldn't get it?. ... am I going to get connected now or not... I'm sure that you can show me the path to get back.

**Reply by zFernand0** [reply]
In the meantime, since you were required to install the Zowe CLI, could you run `zowe zosmf check status` in a terminal and share if you got any errors?

**Reply by 3TInfoTinker** [reply]
"Could you confirm that your profile is still under **~/.zowe/profiles/**" I used this command and opened it in a new window:

![image](https://user-images.githubusercontent.com/74275625/101912148-2d62d980-3be3-11eb-8501-e77c141fcc10.png)

**Reply by 3TInfoTinker** [reply]
> In the meantime, since you were required to install the Zowe CLI, could you run `zowe zosmf check status` in a terminal and share if you got any errors?

Yes, right away

**Reply by 3TInfoTinker** [reply]
on both terminals, Linux, and vscode:

![image](https://user-images.githubusercontent.com/74275625/101912801-0f49a900-3be4-11eb-9c08-5342028291d7.png)

**Reply by zFernand0** [reply]
do you have NodeJS installed? Run `node --version`

**Comment by jellypuno** [comment]
@3TInfoTinker Is it possible to post your logs here after reloading VSCode? 

To reload: View > Command Palette > Developer: Reload Window
Logs can be found here: C:\Users\YourUserName\.vscode\extensions\zowe.vscode-extension-for-zowe-1.10.1\logs

**Reply by 3TInfoTinker** [reply]
I'm using Linux/ubuntu, - and on its 0Level of learning.
It looks like you are asking for win-climate

**Reply by 3TInfoTinker** [reply]
Whenever tried to install Zowe CLI, it talked back like this: 

![image](https://user-images.githubusercontent.com/74275625/101915579-9c423180-3be7-11eb-904f-b274670f2d5c.png)

**Reply by zFernand0** [reply]
at this point the CLI was successfully installed, that last failure you got is for validating plugins.
try `zowe--version`

**Reply by zFernand0** [reply]
the linux equivalent to what `jellypuno` described should be `~/.vscode/extensions/zowe.vscode-extension-for-zowe-1.10.1/logs`

**Reply by 3TInfoTinker** [reply]
Looks like not installed yet

![image](https://user-images.githubusercontent.com/74275625/101916456-c6482380-3be8-11eb-9312-c55182be1cfa.png)

**Reply by 3TInfoTinker** [reply]
at last,  I can see some files, a file with name   
CHANGELOG.md
log4jsconfig.json
package.json
package.nls.json
readmeme.md
theia_tests.jenkinsfile
zowe-vscode-extension.ppf
licence.txt

now which file to share and how to get it copied

**Reply by jellypuno** [reply]
Is there a logs folder? you can upload here the zowe.log and imperative.log

**Reply by 3TInfoTinker** [reply]
this is how the list displays

![image](https://user-images.githubusercontent.com/74275625/101917553-2f7c6680-3bea-11eb-89e8-f35e9bfe637a.png)

**Reply by jellypuno** [reply]
🤔 weird...

So you've installed node and npm and then vscode + zowe extension. When you open it nothing happens and it seems like the logs are not even created.

**Reply by 3TInfoTinker** [reply]
File not found:

![image](https://user-images.githubusercontent.com/74275625/101918501-3fe11100-3beb-11eb-9f9c-30c047ca36bc.png)

**Reply by 3TInfoTinker** [reply]
Not found Zow.log: folder or file
![image](https://user-images.githubusercontent.com/74275625/101918748-89316080-3beb-11eb-8e6a-c4edc4b37486.png)

**Reply by 3TInfoTinker** [reply]
for node and npm:

![image](https://user-images.githubusercontent.com/74275625/101918879-baaa2c00-3beb-11eb-862f-25c24eb719bb.png)

**Reply by 3TInfoTinker** [reply]
You, people, know well about the system and code,
is there anything else that can interact with all this? I mean if the chrome browser was using Y! engine, and then I changed it to Google?
just a thought that this "**_yaml_**" and **_Y!_** looks like Yahoo search engine🙊

**Comment by zFernand0** [comment]
For resolving the CLI instalation problem, (`EACCES`) Please refer to 
https://docs.zowe.org/stable/troubleshoot/cli/known-cli.html#eaccess-error-when-issing-npm-install-command

**Reply by 3TInfoTinker** [reply]
Ok, and thank you for taking such a big headache for me. Lots of thanks and I am really grateful.💝
